# EDA på PLN dataset

In [2]:
import polars as pl

In [3]:

df_images = pl.read_parquet('../data/processed/labels/df_images.parquet')
df_annotations = pl.read_parquet('../data/processed/labels/df_annotations.parquet')
df_categories = pl.read_parquet('../data/processed/labels/df_categories.parquet')
df_full = pl.read_parquet('../data/processed/labels/df_full.parquet')


In [5]:
df_full.describe()

shape: (9, 13)
┌────────────┬───────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬─────────┐
│ statistic  ┆ segmentat ┆ area      ┆ iscrowd   ┆ … ┆ height    ┆ width     ┆ supercate ┆ name    │
│ ---        ┆ ion       ┆ ---       ┆ ---       ┆   ┆ ---       ┆ ---       ┆ gory      ┆ ---     │
│ str        ┆ ---       ┆ f64       ┆ f64       ┆   ┆ f64       ┆ f64       ┆ ---       ┆ str     │
│            ┆ f64       ┆           ┆           ┆   ┆           ┆           ┆ str       ┆         │
╞════════════╪═══════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪═════════╡
│ count      ┆ 3.263046e ┆ 3.263046e ┆ 3.263046e ┆ … ┆ 3.263046e ┆ 3.263046e ┆ 3263046   ┆ 3263046 │
│            ┆ 6         ┆ 6         ┆ 6         ┆   ┆ 6         ┆ 6         ┆           ┆         │
│ null_count ┆ 0.0       ┆ 0.0       ┆ 0.0       ┆ … ┆ 0.0       ┆ 0.0       ┆ 0         ┆ 0       │
│ mean       ┆ null      ┆ 25094.246 ┆ 0.0       ┆ … ┆ 792.47446 ┆ 595.18544 ┆ null      ┆ null    │
│            ┆           ┆ 512       ┆           ┆   ┆ 1         ┆ 5         ┆           ┆         │
│ std        ┆ null      ┆ 33536.571 ┆ 0.0       ┆ … ┆ 30.800791 ┆ 28.876088 ┆ null      ┆ null    │
│            ┆           ┆ 138       ┆           ┆   ┆           ┆           ┆           ┆         │
│ min        ┆ null      ┆ 0.0       ┆ 0.0       ┆ … ┆ 431.0     ┆ 391.0     ┆           ┆ figure  │
│ 25%        ┆ null      ┆ 3707.921  ┆ 0.0       ┆ … ┆ 791.0     ┆ 596.0     ┆ null      ┆ null    │
│ 50%        ┆ null      ┆ 15076.477 ┆ 0.0       ┆ … ┆ 792.0     ┆ 596.0     ┆ null      ┆ null    │
│            ┆           ┆ 708       ┆           ┆   ┆           ┆           ┆           ┆         │
│ 75%        ┆ null      ┆ 33871.302 ┆ 0.0       ┆ … ┆ 794.0     ┆ 601.0     ┆ null      ┆ null    │
│            ┆           ┆ 485       ┆           ┆   ┆           ┆           ┆           ┆         │
│ max        ┆ null      ┆ 397042.56 ┆ 0.0       ┆ … ┆ 1224.0    ┆ 855.0     ┆           ┆ title   │
│            ┆           ┆ 7074      ┆           ┆   ┆           ┆           ┆           ┆         │
└────────────┴───────────┴───────────┴───────────┴───┴───────────┴───────────┴───────────┴─────────┘

In [6]:
df_full.head()

shape: (5, 12)
┌────────────────┬────────────────┬─────────┬──────────┬───┬────────┬───────┬───────────────┬──────┐
│ segmentation   ┆ area           ┆ iscrowd ┆ image_id ┆ … ┆ height ┆ width ┆ supercategory ┆ name │
│ ---            ┆ ---            ┆ ---     ┆ ---      ┆   ┆ ---    ┆ ---   ┆ ---           ┆ ---  │
│ list[list[deci ┆ decimal[38,14] ┆ i64     ┆ i64      ┆   ┆ i64    ┆ i64   ┆ str           ┆ str  │
│ mal[38,2]]]    ┆                ┆         ┆          ┆   ┆        ┆       ┆               ┆      │
╞════════════════╪════════════════╪═════════╪══════════╪═══╪════════╪═══════╪═══════════════╪══════╡
│ [[52.38,       ┆ 13787.43308252 ┆ 0       ┆ 0        ┆ … ┆ 811    ┆ 613   ┆               ┆ text │
│ 444.87, …      ┆ 503600         ┆         ┆          ┆   ┆        ┆       ┆               ┆      │
│ 444.87]]       ┆                ┆         ┆          ┆   ┆        ┆       ┆               ┆      │
│ [[309.91,      ┆ 3266.252633666 ┆ 0       ┆ 0        ┆ … ┆ 811    ┆ 613   ┆               ┆ text │
│ 444.91, …      ┆ 96900          ┆         ┆          ┆   ┆        ┆       ┆               ┆      │
│ 444.91]]       ┆                ┆         ┆          ┆   ┆        ┆       ┆               ┆      │
│ [[52.38,       ┆ 8037.095046702 ┆ 0       ┆ 0        ┆ … ┆ 811    ┆ 613   ┆               ┆ text │
│ 507.66, …      ┆ 65600          ┆         ┆          ┆   ┆        ┆       ┆               ┆      │
│ 507.66]]       ┆                ┆         ┆          ┆   ┆        ┆       ┆               ┆      │
│ [[321.87,      ┆ 18566.00016906 ┆ 0       ┆ 0        ┆ … ┆ 811    ┆ 613   ┆               ┆ text │
│ 465.80, …      ┆ 159000         ┆         ┆          ┆   ┆        ┆       ┆               ┆      │
│ 465.80]]       ┆                ┆         ┆          ┆   ┆        ┆       ┆               ┆      │
│ [[40.42,       ┆ 17140.38018086 ┆ 0       ┆ 0        ┆ … ┆ 811    ┆ 613   ┆               ┆ text │
│ 397.09, …      ┆ 704000         ┆         ┆          ┆   ┆        ┆       ┆               ┆      │
│ 397.09]]       ┆                ┆         ┆          ┆   ┆        ┆       ┆               ┆      │
└────────────────┴────────────────┴─────────┴──────────┴───┴────────┴───────┴───────────────┴──────┘

In [12]:
df_full['file_name'].n_unique()

335703

335703 unika men har 47958 dokument. Behöver alltså minska df mer.

## Att behålla beroende på syfte
### Binär klassificering av förstasida eller inte
Behåll inget